wgpu-hal/src/metal/adapter.rs

```diff
@@ -57,12 +57,12 @@ impl crate::Adapter<super::Api> for super::Adapter {
             }
         };
         let msaa_desktop_if = if pc.msaa_desktop {
-            Tfc::MULTISAMPLE
+            Tfc::MULTISAMPLE_X4
```

Comment:
We need to call https://developer.apple.com/documentation/metal/mtldevice/1433355-supportstexturesamplecount and store it in our private capabilities and use that to determine the multisample levels supported.

Reply:
Do we still need to check for pc.msaa_desktop and pc.msaa_apple7 or can we just use the flag like so : 
`    let msaa_desktop_if = pc.sample_count_mask;`
     `let msaa_apple7x_if =pc.sample_count_mask;
  `?

Reply:
Probably calling `supportstexturesamplecount` is probably good enough and we don't need a general flag.